system_metrics_collector/src/system_metrics_collector/main.cpp

```diff
@@ -67,12 +76,22 @@ int main(int argc, char ** argv)
     RCUTILS_LOG_ERROR_NAMED("main", "Unable to set debug logging for the memory node");
   }
 
+  r = rcutils_logging_set_logger_level(process_mem_node->get_name(), RCUTILS_LOG_SEVERITY_DEBUG);
+
+  if (r != 0) {
+    RCUTILS_LOG_ERROR_NAMED("main", "Unable to set debug logging for the process memory node");
+  }
```

Comment:
Consider doing that instead:

```c++
  {
    const auto r = rcutils_logging_set_logger_level(process_mem_node->get_name(), RCUTILS_LOG_SEVERITY_DEBUG);
    if (r != 0) {
      RCUTILS_LOG_ERROR_NAMED("main", "Unable to set debug logging for the process memory node");
    }
  }
```

+ we should always print out the error string related to `r` =>  see`rcutils_get_error_string()`.
There may be a macro in `rcutils` to do the logging end-to-end.